sdk/cosmosdb/cosmos/src/documents/PartitionKey.ts

```diff
@@ -1,5 +1,50 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT license.
-import { PartitionKeyDefinition } from "./PartitionKeyDefinition";
 
-export type PartitionKey = PartitionKeyDefinition | string | number | unknown;
+import { NonePartitionKeyLiteral, NullPartitionKeyLiteral } from "./PartitionKeyInternal";
+
+export type PartitionKey = PrimitivePartitionKeyValue | PrimitivePartitionKeyValue[];
```

Comment:
This would be technically a breaking change is this is used as input. `unknown` would allow any object, and with the change, this is being narrowed down.

For example, before I could do 

```ts
const partition: PartitionKey = {foo: "bar"};
```

with this change, the above will break.

However, it would depend on how the service (or even the client) currently handles such a scenario. If the scenario is currently broken, meaning, anyone using the example above will get a failure all the time. Then we can consider this a fix and not a breaking change.

On the other hand, if users can make a successful call with a partition key similar to the example above, then this is considered a breaking change.

@v1k1 I'd defer the analysis to you as you are way more familiar with the service than I am. Please help me answer this question

What happens today if the users sets a partitionKey with a value similar to `{ foo: "bar"}`?

Reply:
You guessed it right, even though it seems breaking change, it actually is not since the proper shape of `partitionKey` was always enforced from backend. i.e. Even though there would have been no type error at client code for `{foo: "bar"}`, it would have failed at server. So by trial and error, all existing users must be using correct shape of partition key.